### PR TITLE
add cosmetic bobbing setting

### PR DESCRIPTION
**Pull request by rfomin**
Taken from DSDA-Doom.

### CODE REVIEW

File: Source/m_menu.c [hunk 1]
```diff
@@ -3021,7 +3014,6 @@ static const char *weapon_attack_alignment_strings[] = {
 setup_menu_t weap_settings1[] =  // Weapons Settings screen       
 {
   {"ENABLE RECOIL", S_YESNO,m_null,WP_X, WP_Y+ weap_recoil*8, {"weapon_recoil"}},
-  {"ENABLE BOBBING",S_YESNO,m_null,WP_X, WP_Y+weap_bobbing*8, {"player_bobbing"}},
```
Comment (fabiangreffrath) [inline Source/m_menu.c:3024]: So, this leaves Boom recoil as the only non-cosmetic option in this menu? Hm... do you still have the branch that removes it? 🙇

File: Source/m_misc.c [hunk 2]
```diff
@@ -265,6 +265,13 @@ default_t defaults[] = {
     "1 to enable player bobbing (view moving up/down slightly)"
   },
 
+  {
+    "cosmetic_bobbing",
+    (config_t *) &cosmetic_bobbing, NULL,
+    {1}, {0,2}, number, ss_weap, wad_no,
+    "1 to enable cosmetic player (view moving up/down slightly)"
```
Comment (fabiangreffrath) [inline Source/m_misc.c:272]: I believe there is a word missing here.

File: Source/p_pspr.c [hunk 3]
```diff
@@ -1065,7 +1065,25 @@ void P_MovePsprites(player_t *player)
...
+      last_sy = psp->sy2;
+      psp->sy2 = 32 * FRACUNIT;
+    }
+    else if (psp->state->action == A_Lower)
+    {
+      // We want to move smoothly from where we were
+      psp->sy2 -= (last_sy - 32 * FRACUNIT);
+    }
+  }
+  else if (psp->state && center_weapon)
```
Comment (fabiangreffrath) [inline Source/p_pspr.c:1086]: If cosmetic bobbing and centered weapon sprite exclude each other, we should make this more obvious in the menu.

File: Source/m_misc.c (hunk 2 above)
Comment (fabiangreffrath) [reply Source/m_misc.c:272]: Also, I think `1` as a upport bound.

File: Source/m_menu.c (hunk 1 above)
Comment (rfomin) [reply Source/m_menu.c:3024]: There is also Classic BFG:
doom00

Comment (fabiangreffrath) [reply Source/m_menu.c:3024]: > There is also Classic BFG

Yes, right, that's a special pony. The only hard game-breaking option that we face to the user, right?

File: Source/m_menu.c [hunk 4]
```diff
@@ -2970,7 +2967,8 @@ void M_DrawKeybnd(void)
 // The Weapon Screen tables.
 
 #define WP_X 203
-#define WP_Y  (29+8)
+#define WP_Y  29
+
```
Comment (fabiangreffrath) [inline Source/m_menu.c:2971]: Excess blank line (after rebasing).

File: Source/m_menu.c (hunk 1 above)
Comment (rfomin) [reply Source/m_menu.c:3024]: If you ask me, I'd like to remove Classic BFG and Boom recoil from the menu... and many other options.

File: Source/p_pspr.c (hunk 3 above)
Comment (rfomin) [reply Source/p_pspr.c:1086]: Fixed in https://github.com/fabiangreffrath/woof/pull/528/commits/b1fef3675169126e2acfabc84f76bca0d4eb57ba

File: Source/m_menu.c (hunk 1 above)
Comment (fabiangreffrath) [reply Source/m_menu.c:3024]: I agree the menu is still not perfect, but we have improved a lot already.

I'd like to keep the Classic BFG for now, but agree that "Enable Recoil" has to go.

"Why should I care what I said yesterday" - Konrad Adenauer (roughly) 🤷 😁

**Comment by fabiangreffrath** [comment]
Thank you for your patience! 😉

**Pull request merged**